src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardConsumer.java

```diff
@@ -231,7 +232,8 @@ private static final GetRecordsRetrievalStrategy makeStrategy(KinesisDataFetcher
         this.checkpoint = checkpoint;
         this.recordProcessor = recordProcessor;
         this.recordProcessorCheckpointer = recordProcessorCheckpointer;
-        this.leaseManager = leaseManager;
+        this.leaseManager = leaseCoordinator.getLeaseManager();
```

Comment:
We can avoid having leaseManager as class field and use coordinator to get the value